blockprotocol/blockprotocol
# Update spec + packages to split functions for links from linkedAggregations

We have the concept of two types of 'links' in the protocol:
1. A single target link, which is a traditional link between an entity and another entity, i.e. an edge in a graph
2. A 'linked aggregation', which is a link from a single entity to an 'aggregation' of entities, e.g. some collection of entities which represents the result of some operation/filter.

These were previously defined using the same type and managed using the same operations, which was unwieldy.

This PR splits them out into separate definitions, managed by separate functions. It:
- updates the types in `blockprotocol`
- updates the spec
- updates `mock-block-dock`
- bumps the versions of HASH blocks in the hub

## To test

1. Try out different blocks in the Block Hub.
2. Read the updated spec page for linking entities at `/spec/block-types#linking-entities` and check it makes sense in explaining the two types (links, linkedAggregations)

## Next steps

Update HASH to remove the patch changes introduced in https://github.com/hashintel/hash/pull/497, which should no longer required (a few minor differences to the types here, notably renaming `LinkedAggregationInput` to `LinkedAggregationDefinition`)
